examples/platform/linux/AppMain.cpp

```diff
@@ -150,6 +150,13 @@ int ChipLinuxAppInit(int argc, char ** argv)
         {
             ChipLogError(NotSpecified, "Wi-Fi Management taking too long to start - device configuration will be reset.");
         }
+
+        if (rendezvousFlags.Has(RendezvousInformationFlag::kSoftAP) &&
+            chip::DeviceLayer::ConnectivityMgrImpl().IsWiFiStationProvisioned() == false)
```

Comment:
Neither the condition that the stack just started and the node is not commissioned nor the condition that the stack just started and wifi isn't commissioned are the correct criteria to enable the access point.

There is a state when the commissioning window should be explicitly opened by user or administrative command.  Do we have the ability to key on something like that?  This change here makes a number of non-spec-compliant assumptions.

Reply:
I guess you expect something similar to SetupListModel class for ESP32 platform. Use OpenBasicCommissioningWindow() to start the commissioning procedure.

For Linux we gonna need IPC to trigger these events. For instance create a matter service that listen for external events. DBus might be one possible approach, allowing distros to implement different mechanism to monitor for user commands (eg: Factory reset).

Reply:
@msandstedt It appears the spec allows for commissionable status advertisement upon application of power. This PR appears to assume that app is started at application of power. What criteria should be used to enable SoftAP if that assumption is correct?

The spec requires that advertisement cease 15 minutes after commencement to avoid polluting wireless spectrum. Hence a mechanism by which commissioning window can be explicitly opened needs to be implemented. But these changes perhaps are not what this PR is about.

Reply:
@tewarid,

I see.  However, the correct criteria for enabling the access point for commissioning still should be whether the node is not yet commissioned to a fabric, not IsWiFiStationProvisioned() == false, right?  This seems like a good way to get stuck in a state where you're both uncommissioned and the access point refuses to come up.

Reply:
@msandstedt As code is currently implemented, fabric setting is erased if Wi-Fi (OR Thread) is not provisioned - please see call `OpenBasicCommissioningWindow(ResetFabrics::kYes)` in src/app/server/Server.cpp. That has me thinking that SoftAP should be enabled in OpenBasicCommissioningWindow instead of ChipLinuxAppInit.

Reply:
Why do we erase our fabric if we don't have Wi-Fi or Thread provisioned?  What if we are using ethernet?  I think we need to unwind these couplings.  Wifi or Thread provisioning is a possible (and in some cases necessary) side effect of Matter commissioning, but it is not required.  Commissioning to a Matter fabric means you've received op creds.  This is sometimes accompanied by onboarding to a network, but the former does not always imply the latter and the latter does not always imply the former.

Reply:
@pan-apple Could you clarify why the following call resets fabrics when Wi-Fi or Thread is not provisioned? Do you see any issue with the call happening when device is not commissioned to a fabric instead?

https://github.com/project-chip/connectedhomeip/blob/a0106ab93dea3e15cf73934bf820572e8bee24a7/src/app/server/Server.cpp#L417